symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/model/BdkRetryConfig.java

```diff
@@ -1,28 +1,66 @@
 package com.symphony.bdk.core.config.model;
 
-import lombok.Getter;
+import lombok.NoArgsConstructor;
 import lombok.Setter;
 import org.apiguardian.api.API;
 
-@Getter
 @Setter
+@NoArgsConstructor
 @API(status = API.Status.STABLE)
 public class BdkRetryConfig {
 
-    public static final int DEFAULT_MAX_ATTEMPTS = 10;
-    public static final long DEFAULT_INITIAL_INTERVAL_MILLIS = 500L;
-    public static final double DEFAULT_MULTIPLIER = 2;
-    public static final long DEFAULT_MAX_INTERVAL_MILLIS = 5*60*1000;
-
-    private int maxAttempts = DEFAULT_MAX_ATTEMPTS;
-    private long initialIntervalMillis = DEFAULT_INITIAL_INTERVAL_MILLIS;
-    private double multiplier = DEFAULT_MULTIPLIER;
-    private long maxIntervalMillis = DEFAULT_MAX_INTERVAL_MILLIS;
-
-    public double getMultiplier() {
-        if (this.multiplier < 1) {
-            return 1;
-        }
-        return this.multiplier;
+  public static final int INFINITE_MAX_ATTEMPTS = -1;
+  public static final int DEFAULT_MAX_ATTEMPTS = 10;
+  public static final long DEFAULT_INITIAL_INTERVAL_MILLIS = 500L;
+  public static final double DEFAULT_MULTIPLIER = 2;
+  public static final long DEFAULT_MAX_INTERVAL_MILLIS = 5 * 60 * 1000;
+
+  private Integer maxAttempts;
+  private Long initialIntervalMillis;
+  private Double multiplier;
+  private Long maxIntervalMillis;
+
+  public BdkRetryConfig(Integer maxAttempts) {
+    this.maxAttempts = maxAttempts;
+  }
+
+  public Integer getMaxAttempts() {
+
+    if (this.maxAttempts == null) {
+      return DEFAULT_MAX_ATTEMPTS;
+    }
+
+    if (this.maxAttempts < 0) {
+      // negative value means infinite number of attempts
+      return Integer.MAX_VALUE;
```

Comment:
Does it last until the sun dies? (well unless Elon gets us out of the solar system)

Reply:
Haha, with 2s between each attempt that would give us 8171 years to fix a pod failure :)